projects/components/src/lib/button.directive.ts

```diff
@@ -0,0 +1,65 @@
+import { Directive, ElementRef, HostBinding, HostListener, Input, OnChanges, OnInit, Renderer2, SimpleChanges } from '@angular/core';
+
+@Directive({
+    selector: '[mzButton]',
+    host: {
+        role: 'button',
+        '[attr.tabindex]': '0',
+        '[attr.aria-disabled]': 'busy',
+    },
+})
+export class ButtonDirective implements OnChanges {
+    @Input()
+    variant? = 'secondary';
+    @Input()
+    busy?: boolean;
+    private loadingElement?: Node;
+    private svgContent = `
```

Comment:
what happens if you just add here <mz-spinner> instead of <svg>?

Reply:
As per request from Rui, we can't add this depedency